app/services/reinstate_conditions_met.rb

```diff
@@ -1,59 +1,51 @@
 class ReinstateConditionsMet
-  include ActiveModel::Validations
   include ImpersonationAuditHelper
 
-  attr_reader :application_choice, :course_option
-
-  validates :course_option, presence: true
-  validate :validate_course_option_is_open_on_apply
-  validate :validate_course_option_belongs_to_current_cycle
+  attr_reader :actor, :application_choice, :course_option
 
   def initialize(actor:, application_choice:, course_option:)
-    @auth = ProviderAuthorisation.new(actor: actor)
+    @actor = actor
     @application_choice = application_choice
     @course_option = course_option
   end
 
-  def save
-    @auth.assert_can_make_decisions!(application_choice: application_choice, course_option_id: @course_option.id)
-
-    audit(@auth.actor) do
-      if valid?
-        new_recruited_at = if application_choice.status_before_deferral == 'recruited'
-                             application_choice.recruited_at # conditions are 'still met'
-                           else
-                             Time.zone.now
-                           end
+  def save!
+    auth.assert_can_make_decisions!(application_choice: application_choice,
+                                    course_option: course_option)
 
+    if deferred_offer.valid?
+      audit(actor) do
         ActiveRecord::Base.transaction do
           ApplicationStateChange.new(application_choice).reinstate_conditions_met!
 
           application_choice.update_course_option_and_associated_fields!(
-            @course_option,
-            other_fields: { recruited_at: new_recruited_at },
+            course_option,
+            other_fields: { recruited_at: recruited_at },
           )
           application_choice.offer.conditions.each(&:met!)
-          CandidateMailer.reinstated_offer(application_choice).deliver_later
         end
+        CandidateMailer.reinstated_offer(application_choice).deliver_later
       end
+    else
+      raise ValidationException, deferred_offer.errors.map(&:message)
     end
-  rescue Workflow::NoTransitionAllowed
```

Comment:
have we lost this validation error message being added in the refactor? same below

Reply:
We did, but it was not used anywhere as [the controller calling the service](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/provider_interface/reconfirm_deferred_offers_controller.rb#L50) checked if the service call returned true or false and did not display any of the model errors.

The plan is to handle that in the controller, similar to Make and Change offer, as the error message was put in place specifically for showing it to vendors e.g. https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/vendor_api/decisions_controller.rb#L78

Reply:
Related trello card: https://trello.com/c/u06H8NSO/275-present-errors-when-api-clients-try-to-make-invalid-state-transitions